### PR TITLE
Milestone #2 - Components 

### PR DESCRIPTION
**Pull request by SarvarKh**
## In this milestone, I have:
- [x] prepared the directory structure for components;
- [x] implemented App, Display, Button, ButtonPanel components;
- [x] updated readme.md file;
- [x] added missing MIT.md license file and its content;
- [x] fixed all linter errors;

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function Button(props) {
+  const { name } = props;
+  return (
+    <>
+      Button name:
+      {name}
+    </>
```
Comment (afanimartin) [inline src/components/Button.js]: - This should be an actual HTML button component.
```react
<>
<button>...</button>
</>
```

File: src/components/Button.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  );
+}
+
+Button.propTypes = {
+  name: PropTypes.string,
+};
+
+Button.defaultProps = {
+  name: '0',
+};
```
Comment (afanimartin) [inline src/components/Button.js:21]: - I don't think this validation is needed. You may remove it to make your codebase more clean.

File: src/components/ButtonPanel.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        <Button name="6" />
+        <Button name="-" />
+      </div>
+      <div>
+        <Button name="1" />
+        <Button name="2" />
+        <Button name="3" />
+        <Button name="+" />
+      </div>
+    </>
```
Comment (afanimartin) [inline src/components/ButtonPanel.js:36]: - You missed adding the last group of operators:
```react
<div>
    <Button name="0" />
    <Button name="." />
    <Button name="=" />
 </div>
```

**Review by afanimartin — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 🙏🏿 🙏🏿 
Nice work on this milestone. You managed to meet most of the requirements for this milestone. Bravo!

### Highlights 🌵 🌵 🌵 
- `App`, `Display`, and `DisplayPanel` well implemented.
### Change(s) 🔕 🔕 🔕 
- Check comments for suggestions


### Martin Afani
#### Slack: whiz25

File: src/components/Button.js (hunk 1 above)
Comment (SarvarKh) [reply src/components/Button.js]: Done

File: src/components/Button.js (hunk 2 above)
Comment (SarvarKh) [reply src/components/Button.js:21]: Yes, I agree. However, ESLint was detecting an error if I remove setting default value.
So, I had to **Validate** and give a `defaultProps` value.
Please see below snapshot for more details:
Screen Shot 2021-07-12 at 15 38 38

Also, when I removed **Validation** and `defaultProps`, ESLint is also detecting additional error:
Screen Shot 2021-07-12 at 15 45 33

File: src/components/ButtonPanel.js (hunk 3 above)
Comment (SarvarKh) [reply src/components/ButtonPanel.js:36]: **Indeed, Thanks:+1:**
Done

**Review by rukundoeric — COMMENTED** [review]
# Status: 🍾 PROJECT APPROVED 🍾
_Excellent job on this project guys. 🎆 Keep it up._

- ### To Highlight
  - Follows gitwork-flow.  💯
  - Codebase looks clean and professional. 💯 
  - All changes required from the previous review are addressed. 💯 
  

- ### To Improve
  - N/A

_Congratulations on the splendid job **@team** 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/),
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/),
Slack: Rukundo Eric**

**Comment by SarvarKh** [comment]
## Thank you @rukundoeric.
I couldn't found any comment to work on in your comment above.
Please inform me what exactly I shall change since it is marked as **"Changes Required"**.

**Comment by SarvarKh** [comment]
### @rukundoeric 
I was going to submit for another review, but the dashboard code review request button took me to the "Project completion Form".
It seems the project was marked in the system as approved, but in the above comment was a typo since there is no comment as well. So, I will merge this PR and proceed with next MileStone.

Thank you and regards,
Sarvar

**Comment by rukundoeric** [comment]
@SarvarKh Yeah, I made a mistake while writing the comment. the project is approved. I wish you all the very best in your next project.

**Pull request merged**